### DISCUSSION TITLE
Error during WebSocket handshake: 'Connection' header is missing

### DISCUSSION THREAD

**Discussion by tamagoez** [supabase/supabase#5856] [Questions] [answered]
When I tried to use `.subscribe` option, `failed: Error during WebSocket handshake: 'Connection' header is missing` error occur.
RLS will not affect.
Here is my code.
[Link](https://github.com/tamagoez/sessions/blob/f514b6c0d5c8da940f2e3be004d6ce865a39ad9d/src/lib/Message.js#L10)
```
useEffect(() => {
    supabase
      .from('channels_chat')
      .on('INSERT', handleInsert)
      .subscribe()
}, [])
```
Thank you.

**Comment by GaryAustin1** [comment]
That error implies an issue getting connected to the realtime server.  Doubt it has anything to do with the code you show.
Are you running your own instance of Supabase? 
You might also look at the network tab for websockets and see if there is an error.

**Reply by tamagoez** [reply]
> **You might also look at the network tab for websockets and see if there is an error.**

When I check Network tab, all websocket connection's status was set to `Finished`.(still throw error)
> **Are you running your own instance of Supabase?**

I can't understand meaning of `own instance of Supabase`.
Sorry for my few knowledge.
I don't want to trouble you, but could you explain it please?

**Reply by GaryAustin1** [reply]
Own instance means you have set up your own Supabase server versus using their hosting service.

**Reply by tamagoez** [reply]
Now I get the point.
I'm not using own server, but `app.supabase.io`.

**Reply by GaryAustin1** [reply]
I see supabase.js at 1.29.1 in your package.lock file can you upgrade that to latest version.  
Also if I remember react correctly your useEffect is going to run and setup a new subscription everytime you get a message and update “mes”

**Reply by tamagoez** [reply]
I edited it to `"@supabase/supabase-js": "^1.31.1",` and deployed, but it still throw error.

And thank you for your advice.
Yes, it was my stupid mistake.
Now, I'm working to edit around there.

**Reply by GaryAustin1** [reply]
At this point, I don't see anything wrong in your actual subscribe code (it is pretty simple). Hard to do more without seeing the network log and websocket connection details.  Even then not sure I could find it.  I am assuming your other supabase calls all worked (although they don't use websockets).
I suspect it is something in your react environment with websockets or your supabase instance has an issue with realtime.   You could reach out to support with your supabase url, the error message  you are getting and see if they will take a look at it.

**Reply by tamagoez** [reply]
Is sending public URL safe?

#### Additional Information
**Chrome's Log**
`browser.js:25`
```
WebSocket connection to 'wss://XXXXXXX.supabase.co/realtime/v1/websocket?apikey=OOOOOOO&vsn=1.0.0' failed: Error during WebSocket handshake: 'Connection' header is missing
```

| i | @ | browser.js:25 |
| --- | --- | --- |
| value | @ | RealtimeClient.ts:124 |
| (anonymous) | @ | RealtimeClient.ts:112 |
| c | @ | runtime.js:63 |
| (anonymous) | @ | runtime.js:294 |
| (anonymous) | @ | runtime.js:119 |
| i | @ | serializer.ts:4 |

#### `browser.js`
(line 18 to 39)
``` javascript
/**
 * Expose a W3C WebSocket class with just one or two arguments.
 */
function W3CWebSocket(uri, protocols) {
	var native_instance;

	if (protocols) {
		native_instance = new NativeWebSocket(uri, protocols);
                ^ Error occurred in this line
	}
	else {
		native_instance = new NativeWebSocket(uri);
	}

	/**
	 * 'native_instance' is an instance of nativeWebSocket (the browser's WebSocket
	 * class). Since it is an Object it will be returned as it is when creating an
	 * instance of W3CWebSocket via 'new W3CWebSocket()'.
	 *
	 * ECMAScript 5: http://bclary.com/2004/11/07/#a-13.2.2
	 */
	return native_instance;
}
```

**Comment by GaryAustin1** [comment]
At this point support is still your next step.
There is something going wrong with either your environment or theirs with websockets.
https://app.supabase.io/support/new

**Comment by tamagoez** [comment, marked as answer]
It seems my Anti Virus software's setting affect to it.
When I tried [This](https://github.com/shwosner/realtime-chat-supabase-react), same error happend.
But I don't know why it's affect.
Sorry for @GaryAustin1 and also Supabase Support Team that it was my fault.
When I hold a conviction, I will close this dicusussion.

**Comment by GaryAustin1** [comment]
@tamagoez Could you provide the name of your anti virus software?  If anti virus software can block websockets, by default especially, a note in the documents at supabase might be helpful.

**Reply by tamagoez** [reply]
I'm using i-Filter to protect.
It's actually filtering service what can protect virus from web.
It seems it affect to Websocket connection, too.
So I may need to change some setting, or use other computer.
